src/pack_unpack.c

```diff
@@ -798,6 +814,10 @@ json_t *json_vpack_ex(json_error_t *error, size_t flags,
         set_error(&s, "<format>", json_error_invalid_format, "Garbage after format string");
         return NULL;
     }
+    if(s.scan_mode && value) {
```

Comment:
The `&& value` is technically not needed, because it's already checked a few lines up.

Reply:
I've removed the unneeded check here and a few other places where it was not needed.

Reply:
:+1: